src/Build.UnitTests/Microsoft.Build.Engine.UnitTests.csproj

```diff
@@ -249,6 +249,7 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup Condition="'$(NetCoreBuild)' != 'true'">
+    <Reference Include="System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
```

Comment:
I've had to add this reference in order to build the project. Does anyone know why?

Reply:
I think it's because ImmutableArray got exposed to the tests (FileMatcher_Tests, via the new cache). Then FileMatcher is included in two test dlls.

Reply:
I understand that the two test dlls need the dependency, what I don't understand is that adding the dependency to the `project.json` don't work for `Engine.UnitTests`, but [it works](https://github.com/Microsoft/msbuild/pull/2572/commits/1322acf14ed4e26a88bd3bcc927ea13b94426dd0#diff-c06dddefcbd40b425acebaac4b74d20eR15) for the `Tasks.UnitTests` project. Its seems like that one of the `Engine.UnitTests` dependency needs a lower version of the Immutable dll, but from the logs it is not clear which one.